src/ev3_bluetooth.c

```diff
@@ -0,0 +1,332 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <poll.h>
+#include <ctype.h>
+#include "../firmware_headers/bluetooth/bluetooth.h"
+#include "../firmware_headers/bluetooth/hci.h"
+#include "../firmware_headers/bluetooth/hci_lib.h"
+#include "../firmware_headers/bluetooth/rfcomm.h"
+
+#include "ev3_bluetooth.h"
+
+/**
+ * Channel 1 is already taken by the UI process of the robot, so we use 2.
+ */
+#define DEFAULT_C4EV3_RFCOMM_CHANNEL    2
+#define MAX_CONNECTIONS                 7
+#define SHORT_BLUETOOTH_ADDRESS_LENGTH  sizeof("000000000000")
+
+#define KNOWN_BLUETOOTH_NAMES_FILE_NAME_LENGTH 256
+#define KNOWN_BLUETOOTH_NAMES_FILE_ROW_LENGTH  256
+
+void BluetoothInit () { }
+
+BluetoothConnectionHandle connections[MAX_CONNECTIONS];
+int currentConnections = 0;
+
+void addConnectionToList (BluetoothConnectionHandle c) {
+	if (currentConnections > MAX_CONNECTIONS) {
+		return;
+	}
+	connections[currentConnections++] = c;
+}
+
+void removeConnectionFromList (BluetoothConnectionHandle c) {
+	int i;
+	for (i = 0; i < currentConnections; i++) {
+		if (connections[i] == c) {
+			break;
+		}
+	}
+	if (i >= currentConnections) {
+		return;
+	}
+	currentConnections--;
+	for (; i < currentConnections; i++) {
+		connections[i] = connections[i + 1];
+	}
+}
+
+
+bool isBluetoothAddress(const char * nameOrAddress);
+bool findAddressByBluetoothName(const char * name, char * address);
+
+/**
+ * Connects to the specified bluetooth address. If the connection fails, a new
+ * attempt is tried after one second.
+ * @param address
+ * @param attempts number of attempts to retry connection
+ * @return connection handle or -1 if the connection couldn't be established
+ */
+BluetoothConnectionHandle connectByBluetoothAddress(const char * address, int attempts);
+
+BluetoothConnectionHandle ConnectTo(const char * nameOrAddress, int attempts) {
+	char address[BLUETOOTH_ADDRESS_LENGTH];
+	if (isBluetoothAddress(nameOrAddress)) {
+		strncpy(address, nameOrAddress, BLUETOOTH_ADDRESS_LENGTH);
+		address[BLUETOOTH_ADDRESS_LENGTH - 1] = '/0';
```

Comment:
`\0`. That should have given you a warning about t being an unportable multicharacter literal.

Reply:
Fixed :+1:   5f5d148